pandas/tests/groupby/test_function.py

```diff
@@ -1594,3 +1594,19 @@ def test_multiindex_group_all_columns_when_empty(groupby_func):
     result = method(*args).index
     expected = df.index
     tm.assert_index_equal(result, expected)
+
+
+def test_deprecated_keywords():
+    msg = (
+        "In the future version of pandas the arguments args"
+        "and kwargs will be deprecated for these functions"
+    )
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        df = DataFrame({"a": [1, 1, 2], "b": [3, 4, 5]})
+        gb = df.groupby("a")
+        assert gb.cummax(kwargs=1)
+        assert gb.cummin(kwargs=1)
+        assert gb.cumsum(args=1, kwargs=1)
```

Comment:
I think you've got this wrong. This is not testing `args` at all, to test if the `*args` deprecation is working you need to use `gb.cumsum(1, incorrect_argument=1)`. And I'd recommend splitting this into two separate tests, otherwise you will only test that one of them works, not that both work.

I'd recommend that you make sure you understand well what the `*` and `**` do in a function signature like `def foo(my_arg, *args, **kwargs):`